src/nci_core.c

```diff
@@ -297,53 +294,65 @@ nci_core_closure_cb(
     NciCoreObject* self,
     NciCoreClosure* closure)
 {
-    closure->func(&self->core, closure->user_data);
+    closure->func.core(&self->core, closure->user_data);
 }
 
 static
 void
 nci_core_intf_activated_closure_cb(
     NciCoreObject* self,
     const NciIntfActivationNtf* ntf,
-    NciCoreIntfActivationClosure* closure)
+    NciCoreClosure* closure)
 {
-    closure->func(&self->core, ntf, closure->user_data);
+    closure->func.intf_activation(&self->core, ntf, closure->user_data);
 }
 
 static
 void
 nci_core_data_packet_closure_cb(
     NciCoreObject* self,
     guint8 cid,
-    const void* payload,
+    const void* pkt,
     guint len,
-    NciCoreDataPacketClosure* closure)
+    NciCoreClosure* closure)
 {
 #pragma message("Use GUtilData for payload?")
-    closure->func(&self->core, cid, payload, len, closure->user_data);
+    closure->func.data_packet(&self->core, cid, pkt, len, closure->user_data);
+}
+
+static
+void
+nci_core_param_change_closure_cb(
+    NciCoreObject* self,
+    NCI_CORE_PARAM id,
```

Comment:
Most other places (except some test code) has NCI_CORE_PARAM parameter named as key. Not really critical since these are only internal things but wondering if there should be consistent naming.

Reply:
consistency is a good thing